tests/qos/test_buffer.py

```diff
@@ -0,0 +1,356 @@
+import logging
+import os
+import sys
+import time
+import re
+
+import pytest
+
+
+profile_format = 'pg_lossless_{}_{}_profile'
+
+default_cable_length_list = ['5m', '40m', '300m']
+default_mtu = '9100'
+
+def check_pool_size(duthost, expected_pool_size):
+    pool_size = duthost.shell('redis-cli hget "BUFFER_POOL:ingress_lossless_pool" size')['stdout']
+    assert int(pool_size) == expected_pool_size, "Pool size isn't correct: expected {} but got {}".format(expected_pool_size, pool_size)
+
+
+def check_profile(duthost, pg, expected_profile):
+    profile = duthost.shell('redis-cli hget {} profile'.format(pg))['stdout'][1:-1]
+    assert profile == 'BUFFER_PROFILE:' + expected_profile, 'Expected profile {} not found'.format(expected_profile)
+
+
+def check_pfc_enable(duthost, port, expected_pfc_enable_map):
+    pfc_enable = duthost.shell('redis-cli -n 4 hget "PORT_QOS_MAP|{}" pfc_enable'.format(port))['stdout']
+    assert expected_pfc_enable_map == pfc_enable, \
+        "Expected pfc enable map {} doesn't match {}".format(expected_pfc_enable_map, pfc_enable)
+
+
+def detect_ingress_pool_number(duthost):
+    pools = duthost.shell('redis-cli -n 4 keys "BUFFER_POOL|ingress*"')['stdout']
```

Comment:
stdout_lines could be used, return list. Do not need to user len(pools.split()) -> return len(pools)

Reply:
Hi @SavchukRomanLv ,
Thank you for your comments. But I'm afraid "len(pools)" won't work because it returns the length of pools while it should return the number of lines of pools.
Let me know for further suggestions.
Thanks.